src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAISerializationHelpers.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.ServerSentEvents;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI.Chat;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Defines a set of helpers used to serialize Microsoft.Extensions.AI content using the OpenAI wire format.
+/// </summary>
+public static class OpenAISerializationHelpers
+{
+    /// <summary>
+    /// Deserializes a chat completion request in the OpenAI wire format into a pair of <see cref="ChatMessage"/> and <see cref="ChatOptions"/> values.
+    /// </summary>
+    /// <param name="stream">The stream containing a message using the OpenAI wire format.</param>
+    /// <param name="cancellationToken">A token used to cancel the operation.</param>
+    /// <returns>The deserialized list of chat messages and chat options.</returns>
+    public static async Task<OpenAIChatCompletionRequest> DeserializeChatCompletionRequestAsync(
+        Stream stream, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(stream);
+
+        BinaryData binaryData = await BinaryData.FromStreamAsync(stream, cancellationToken).ConfigureAwait(false);
+        ChatCompletionOptions openAiChatOptions = JsonModelHelpers.Deserialize<ChatCompletionOptions>(binaryData);
```

Comment:
Not really. The `IJsonModel<T>` abstraction used by the Azure models inherently does not support streaming. I experimented with using a pooled `JsonDocument` instead of `BinaryData` but with currently available APIs this forces copying the data to two separate buffers. `BinaryData` seemed simple enough.

Reply:
> The IJsonModel<T> abstraction used by the Azure models inherently does not support streaming

Write accepts a Utf8JsonWriter which can be constructed around a Stream. I assume then you mean it doesn't support _asynchronous_ streaming.

Reply:
But even we were considering serialization, I don't think using Utf8JsonWriter over streams is something we could consider. IIRC aspnetcore streams will throw an exception if you try to invoke any of the sync APIs.